bot/tests/test_issues.py

```diff
@@ -65,3 +68,38 @@ def test_cleaned_paths(log, path, cleaned_path):
         assert log.has(
             "Cleaned issue absolute path", path=path, name="test-task", level="warning"
         )
+
+
+def test_build_hash(mock_revision, mock_hgmo):
```

Comment:
Maybe it would be better to have a separate file where we put all the hash-related tests, as I can see we might have a lot of them over time (as we find improvements to the algorithm).